File: backend/src/main/java/unpsjb/labprog/backend/model/LogValidacionParteMO.java
package unpsjb.labprog.backend.model;

import java.time.LocalDate;
import java.time.LocalTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class LogValidacionParteMO {


    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private int id;

    @Column(nullable=false)
    @JsonFormat(pattern = "yyyy-MM-dd", shape = JsonFormat.Shape.STRING)
    private LocalDate fecha;

    private LocalTime hora;


    @ManyToOne(fetch = FetchType.EAGER)
    private Estado estado;

    @ManyToOne(fetch = FetchType.EAGER)
    private ValidacionParteMO validacionParteMO;

    


    
}

File: frontend/cli/src/app/procesoValidacion/log-validacion-parte-mo.ts
import { Time } from "@angular/common";
import { ParteMO } from "../partes/parteMO";
import { Estado } from "./estado";
import { ValidacionParteMO } from "./validacion-parte-mo";

export interface LogValidacionParteMO {
    id: Number;
    hora: Time;
    estado: Estado;
    validacionParteMO: ValidacionParteMO;
    fecha: Date;
}